### PR TITLE
Bug 1708166 Deploy per-glean app datasets for all apps

### PR DESCRIPTION
**Pull request by jklukas**
We are already creating `app_name` datasets for Fenix and Firefox for iOS
(our multi-app_id apps) via https://github.com/mozilla/bigquery-etl/pull/2027

This extends to all apps per https://bugzilla.mozilla.org/show_bug.cgi?id=1708166#c2

### CODE REVIEW

File: bigquery_etl/glean_usage/events_unnested.py [hunk 1]
```diff
@@ -13,12 +13,9 @@ class EventsUnnestedTable(GleanTable):
...
         self.prefix = PREFIX
-        self.custom_render_kwargs = {}
-        self.no_init = False
+        self.no_init = True
+        self.per_app_id_enabled = False
         self.cross_channel_template = "cross_channel_events_unnested.view.sql"
-
-    def generate_per_app_id(self, project_id, baseline_table, output_dir=None):
-        """Generate the baseline table query per app_name."""
-        logging.info("generate_per_app_id() not implemented for EventsUnnestedTable")
```
Comment (jklukas) [inline bigquery_etl/glean_usage/events_unnested.py:24]: This is now handled by a `per_app_id_enabled` check in the body of the upstream function.

**Comment by jklukas** [comment]
I've run this locally and compared the output via `bqetl glean_usage generate --output-dir tmpsql7` and compared the output:

```
› diff -qrN tmpsql5 tmpsql7 | awk '{print $2}'
tmpsql5/moz-fx-data-shared-prod/bergamot/dataset_metadata.yaml
tmpsql5/moz-fx-data-shared-prod/burnham/events_unnested/view.sql
tmpsql5/moz-fx-data-shared-prod/firefox_desktop/events_unnested/view.sql
tmpsql5/moz-fx-data-shared-prod/firefox_desktop_background_update/events_unnested/view.sql
tmpsql5/moz-fx-data-shared-prod/firefox_echo_show/dataset_metadata.yaml
tmpsql5/moz-fx-data-shared-prod/firefox_fire_tv/dataset_metadata.yaml
tmpsql5/moz-fx-data-shared-prod/firefox_reality/dataset_metadata.yaml
tmpsql5/moz-fx-data-shared-prod/firefox_reality_pc/dataset_metadata.yaml
tmpsql5/moz-fx-data-shared-prod/focus_ios/dataset_metadata.yaml
tmpsql5/moz-fx-data-shared-prod/klar_ios/dataset_metadata.yaml
tmpsql5/moz-fx-data-shared-prod/lockwise_android/dataset_metadata.yaml
tmpsql5/moz-fx-data-shared-prod/lockwise_ios/dataset_metadata.yaml
tmpsql5/moz-fx-data-shared-prod/mach/dataset_metadata.yaml
tmpsql5/moz-fx-data-shared-prod/mozphab/events_unnested/view.sql
tmpsql5/moz-fx-data-shared-prod/mozregression/dataset_metadata.yaml
tmpsql5/moz-fx-data-shared-prod/rally_debug/dataset_metadata.yaml
tmpsql5/moz-fx-data-shared-prod/reference_browser/dataset_metadata.yaml
```

For apps like `burnham` where normalized app_id and app_name are equal, we see `events_unnested` added.

For apps like `bergamot` where the app_id differs, we see `dataset_metadata.yaml` get generated for the new dataset. Once the ops logic goes ahead and creates these, the next run of generation will populate these with the baseline_* views and the events_unnested view.

File: bigquery_etl/glean_usage/common.py [hunk 2]
```diff
@@ -150,7 +150,9 @@ def get_app_info():
     app_info = {}
 
     for app in apps_json:
-        if app["app_name"] not in app_info:
+        if app["app_id"].startswith("rally-"):
```
Comment (jklukas) [inline bigquery_etl/glean_usage/common.py:153]:
```suggestion
        if app["app_id"].startswith("rally"):
```

This was missing `rally.debug`.

**Comment by jklukas** [comment]
cc @whd This will cause 16 new datasets to be created via `dataset_metadata.yaml` files.

**Comment by whd** [comment]
As a transient note, we're having some ops infra issues right now so propagation time for these datasets may be increased (propagation time from generated-sql to mozdata prod is generally 1-2 hours, though I'm looking at speeding that up).

**Pull request merged**